### PR TITLE
Blog post: Monitoring Microservices with OpenShift, Hawkular Metrics …

### PR DESCRIPTION
**Pull request by jotak**
…and Grafana
- [x]  Fix TODO in text

### CODE REVIEW

**Comment by jotak** [comment]
It's WIP but please feel free to start reviewing.
My English style being quite basic feel free also to suggest better formulation whenever needed!

File: src/main/jbake/content/blog/2016/10/17/hawkular-metrics-openshift-and-grafana.adoc [hunk 1]
```diff
@@ -0,0 +1,221 @@
+= Monitoring Microservices with OpenShift, Hawkular Metrics and Grafana
+Joel Takvorian
+2016-10-17
```
Comment (pilhuhn) [inline src/main/jbake/content/blog/2016/10/17/hawkular-metrics-openshift-and-grafana.adoc]: Take a different date - we already have one on 17th

File: src/main/jbake/content/blog/2016/10/17/hawkular-metrics-openshift-and-grafana.adoc [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, metrics, microservice, vertx, openshift, grafana
+
+OpenShift, Hawkular Metrics and Grafana are three great tools that we can combine build a powerful monitoring system for microservices.
```
Comment (pilhuhn) [inline src/main/jbake/content/blog/2016/10/17/hawkular-metrics-openshift-and-grafana.adoc]: combine to build

**Comment by pilhuhn** [comment]
Looks already pretty cool

**Comment by jotak** [comment]
For information this blog post requires 2 things before being submitted:
- PR https://github.com/hawkular/hawkular-grafana-datasource/pull/18 must be merged
- hawkular metrics 0.20.0 (or above) must be available in origin-metrics

**Comment by theute** [comment]
It's indeed very nice ! Congrats !

The first paragraph may scare away a few because of its market-ish wording, but the rest is nice with a lot of useful screenshots !

**Comment by jotak** [comment]
@mwringe, could you check that I'm not writing anything wrong on the openshift config (beginning of the article). Also, at some point in the article I'm writing that the different metrics, like memory /CPU usage etc. are provided by OpenShift, I'm not sure if it's a reasonable shortcut, they come from heapster, right?

**Comment by jotak** [comment]
@theute thank you, well about the first paragraph I'm just trying to define these 3 products very quickly and in a positive way but if it smells too much auto-satisfaction then I should rewrite...

**Comment by mwringe** [comment]
@jotak The metrics are exposed via a cAdvisor endpoint on each OpenShift/Kubernetes node, its Heapster which collects these metrics from the endpoint and pushes them to Hawkular Metrics. I think its a reasonable shortcut to say its coming from OpenShift. I don't think you want to get into things like systemd and how the metrics are actually gathered :)

**Comment by theute** [comment]
"your favorite" scared me a bit :)
"And OpenShift 3 is the new container platform from Red Hat built on top of Docker and Kubernetes."

File: src/main/jbake/content/blog/2016/10/18/hawkular-metrics-openshift-and-grafana.adoc [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+- Hawkular Metrics is the robust and flexible layer that is used for metrics storage and querying.
+
+- And OpenShift is your favorite microservices orchestrator on top of Docker and Kubernetes.
+
+Let's see how to make them work together and visualize our microservices metrics in Grafana.
+
+=== Preparing OpenShift
+
+I assume you already have an access to a running OpenShift Origin (1.0.8 or later), but if you don't, you can link:https://github.com/openshift/origin[follow the installation steps]. Or if you're a lazy boy/girl like me, you can use link:https://github.com/jimmidyson/minishift[MiniShift] to get it running very quickly. Just make sure you assign enough memory and CPU resources to the VM; I could get everything running fine with 2 cores and 4GB, of course that will depend on your hardware.
```
Comment (jshaughn) [inline src/main/jbake/content/blog/2016/10/18/hawkular-metrics-openshift-and-grafana.adoc]: trivial: 'already have an access' -> 'already have access'

**Comment by jotak** [comment]
@mwringe ok, I'll leave it unchanged then. Btw, I've seen that you merged some work on hawkular-metrics, does it mean that the next release of metrics will be included in origin-metrics?

**Comment by jotak** [comment]
@theute ok I will change that!

File: src/main/jbake/content/blog/2016/10/18/hawkular-metrics-openshift-and-grafana.adoc [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+endif::[]
+
+You can play around with OpenShift. Scale some pods up and down, and you'll get the metrics updated in Grafana. Just note that on downscaling, you'll have to wait a little bit (5 minutes) before seeing the _Live_ metrics on the _Singlestat_ panel being updated. This is because we're not sure if the absence of data is due to a pod being shut down, or a simple delay between measurements.
+
+=== It's Just a Beginning
+
+Thanks to the metrics provided in OpenShift, you can build more elaborate dashboards. Just change the _desciptor_name_ tag and see what's interesting for you, there are metrics on memory, CPU, network and filesystem.
+
+But that's just the starting kit! link:http://www.hawkular.org/hawkular-clients/[The Hawkular Metrics ecosystem] is rich and keeps growing, including a Wildfly agent, a Vert.x plugin, a DropWizard reporter, etc. And if that's not enough for you, it's very easy to integrate your own metrics: either through the client libraries or by directly calling the link:http://www.hawkular.org/docs/rest/rest-metrics.html[Metrics REST API].
+
```
Comment (jshaughn) [inline src/main/jbake/content/blog/2016/10/18/hawkular-metrics-openshift-and-grafana.adoc]: if you want to add a teaser you could also mention the upcoming alerting addition to Hmetrics.

Comment (jotak) [reply src/main/jbake/content/blog/2016/10/18/hawkular-metrics-openshift-and-grafana.adoc]: I try but can't find a good formulation, as I'm mostly talking about the metrics API consumers here, Alerts integration seems a bit different, no?

Comment (jshaughn) [reply src/main/jbake/content/blog/2016/10/18/hawkular-metrics-openshift-and-grafana.adoc]: No problem if it doesn't fit the content of the post.   Alerting is now packaged with Metrics so both APIs are available if metrics is deployed, but pure metrics users may not be interested.

**Comment by jotak** [comment]
All required features have been merged (in origin-metrics and in Grafana), so for me it's ready for publishing

**Comment by pilhuhn** [comment]
Thanks @jotak this is great stuff!

**Pull request merged**